x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/Expressions.java

```diff
@@ -21,6 +22,14 @@
 
 public final class Expressions {
 
+    public enum FunctionArgument {
```

Comment:
A more appropriate name would be `ParameterPosition` or `ParamOrdinal` (since it might not be always a function through implementation wise all predicates are).